api/src/main/java/liquibase/ext/datatype/core/PostgreSQLUuidType.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package liquibase.ext.datatype.core;
+
+import liquibase.database.Database;
+import liquibase.database.core.PostgresDatabase;
+import liquibase.datatype.DatabaseDataType;
+import liquibase.datatype.core.CharType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PostgreSQL returns white spaces for UUIDs since their actual length is 36 and we have set it
+ * everywhere as 38 in consideration to ticket https://issues.openmrs.org/browse/TRUNK-1283. This
+ * leads to validation issues. Thus we convert it to CHAR(36).
+ * 
+ * @since 2.4
+ */
+public class PostgreSQLUuidType extends CharType {
+	
+	private static final Logger log = LoggerFactory.getLogger(PostgreSQLUuidType.class);
+	
+	@Override
+	public DatabaseDataType toDatabaseDataType(Database database) {
+		if (database instanceof PostgresDatabase && getSize() == 38) {
```

Comment:
What is it is another char(38) field which is not a uuid?

Reply:
@dkayiwa  @ibacher , I couldn't find a way to get access to column name within the scope of this method. Apparently, what I have understood is we will have to make the changes in diferent change classes [here](https://github.com/liquibase/liquibase/tree/bce767333bd4867708a03ca5fb90df868433fb44/liquibase-core/src/main/java/liquibase/change/core) and this does not seem right.

If we hold a possibility of having any other fields with char(38), what if we apparently make change to using varchar(38) and if we finally are ready to revert https://issues.openmrs.org/browse/TRUNK-1283 , then we make char(36) directly into liquibase changesets for both MySQL and PostgreSQL rather then going this way ?

Reply:
Would the revert require touching all existing changesets?

Reply:
Yes, that is what I have in mind.

Reply:
Would that require a validation checksum for the so many changesets?

Reply:
Apparently, Yes.

Reply:
That would be too many changes.

Reply:
Rather than reverting (which isn't really the way to go) would it make sense to just add a new changeset downsizing the UUID fields?

Reply:
A changeset like this ( I took it from the docs ) : 

```
<changeSet  author="liquibase-docs"  id="modifyDataType-example">  
    <modifyDataType  catalogName="cat"  
            columnName="id"  
            newDataType="int"  
            schemaName="public"  
            tableName="person"/>  
</changeSet>
```
would then need to be added for each table present in core as well as all the ref. application modules. Wouldn't it be adding some overhead ?